src/main/java/racingCar/step4/RacingCars.java

```diff
@@ -0,0 +1,63 @@
+package racingCar.step4;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RacingCars {
+
+    private static final String CAR_NAME_SEPARATOR = ",";
+    private static final int MOVE_CONDITION_LIMIT = 10;
+    private static final Random MOVE_CONDITION_MAKER = new Random();
+
+    private final List<RacingCar> cars;
+
+    private RacingCars(List<RacingCar> cars){
+        this.cars = cars;
+    }
+
+    public static RacingCars createCarsWithNames(String names){
+        List<RacingCar> cars = new ArrayList<>();
+        String[] nameArray = splitName(names);
+        for (String name : nameArray) {
+            RacingCar car = RacingCar.createCarWithName(name);
+            cars.add(car);
+        }
+        return new RacingCars(cars);
+    }
+
+    private static String[] splitName(String names) {
+        return names.split(CAR_NAME_SEPARATOR);
+    }
+
+    public void move() {
```

Comment:
```suggestion
    public void move(Random random) {
```

이렇게 랜덤을 넘겨받으면 제어할 수 없는 것을 외부에서 주입받아 해결할수 있는 장점이 많습니다.

자바의 상속을 활용해 fake 클래스를 만들어 볼게요

```java
public class FakeFixRandom extends Random {
    private int fixedNumber;

    public int nextInt(int bound){
        return fixedNumber;
    }
}
```

위 처럼 fake 를 만들어서 테스트할 때 `move(new FakeFixRandom(5))` 넘기면 모든 차는 전진하고 몇번 전진했는지도 컨트롤이 가능할거에요!

다만, 테스트를 위한 Fake 클래스이니 test scope 에서만 동작하게 패키지를 test 에 두고 사용하는걸 권장드립니다

Reply:
얘기주신 부분 value object로도 가능해보여서
random number를 감싸는 RandomNumber로 클래스 설계해봤습니다